code/drug_synergy/02-drug_synergy_score_calc.R

```diff
@@ -0,0 +1,135 @@
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+
+analysis_dir <- file.path(root_dir, "code", "drug_synergy")
+results_dir <- file.path(analysis_dir, "results", "synergy_score")
+if(!dir.exists(results_dir)){
+  dir.create(results_dir, recursive = TRUE)
+}
+
+source(file.path(analysis_dir, "utils/get_synergy_score.R"))
+
+#### Parse command line options ------------------------------------------------
+
+option_list <- list(
+  make_option(c("-g","--gtex_mapped"),type="character",
+              help="Output file for drug-mapped gtex qSig subsetted subnetworks (.tsv) "),
+  make_option(c("-p","--pbta_mapped"),type="character",
+              help="Output file for drug-mapped pbta qSig subsetted subnetworks (.tsv) "),
+  make_option(c("-h","--pbta_hgg_mapped"),type="character",
+              help="Output file for drug-mapped pbta hgg subsetted subnetworks (.tsv) "),
+  make_option(c("-f","--subnetwork"),type="character",
+               help="File for subnetworks of module of interest (.tsv) "),
+  make_option(c("-b","--output_path"),type="character",
+              help="Path for results output ")
+)
+opt <- parse_args(OptionParser(option_list=option_list,add_help_option = FALSE))
+output_path <- opt$output_path 
+
+#### Read in files necessary for analyses -----------------------------------
+gtex_qSig_subnet_mapped <- readr::read_tsv(opt$gtex_mapped) %>% mutate(comparison = "gtex_qSig")
+pbta_qSig_subnet_mapped <- readr::read_tsv(opt$pbta_mapped) %>% mutate(comparison = "pbta_qSig")
+pbta_hgg_qSig_subnet_mapped <- readr::read_tsv(opt$pbta_hgg_mapped) %>% mutate(comparison = "pbta_hgg_qSig")
+
+subnetwork <- readr::read_tsv(opt$subnetwork)
+
+#### Synergy Score calculated ----------------------------------------------------------
+# following steps adapted from SynGeNet Publication [@doi:10.1038/s41540-019-0085-4]
+# Supplementary information downloadable as `SynGeNet.R`
+
+#### SubNetwork Generation ---------------------------------------------------------------
+subnetwork_graphed <- subnetwork %>% select(Gene1, Gene2) %>% as.matrix() 
+
+  # Induce background network
+subnetwork_graphed <- graph.edgelist(subnetwork_graphed)
+
+# Find the nodes of subnetwork (gene symbols in the subnetworks)
+gene1_names <- subnetwork %>% pull(Gene1) %>% unique()
+gene2_names <- subnetwork %>% pull(Gene2) %>% unique()
+all_nodes <- union(gene1_names, gene2_names)
+
+#### Synergy Score Calculation ---------------------------------------------------------------
+list_of_qSigs <- list(gtex_qSig_subnet_mapped, pbta_qSig_subnet_mapped, pbta_hgg_qSig_subnet_mapped)
+
+# Generate an ordered list of unique drugs (ordered by their WTCS scores)
+lapply(list_of_qSigs, function(x){
+  drug_list <- x %>% 
+    arrange(WTCS) %>%
+    pull(Drug_Name) %>% unique() 
+  nDrug <- length(drug_list)
+  
+  # Give all the drugs weighted score from 1-2 based on their ranking 
+  weight_score <- (1 + (1.0 - (c(1:nDrug)/nDrug)))
```

Comment:
@runjin326 - is this weighting pulled from SynGeNet algorithm?

Reply:
Yes, this `(1 + (1.0 - (c(1:nDrug)/nDrug)))` calculation is from them but it is also slightly different. For them, they ranked the drugs based on how many targets a particular drug has in the subnetwork, take top 100 and give them weights between 2 and 1. For all other drugs that have >=2 targets in the subnetwork, they give them weight of 1. For us, [based on our discussion](https://github.com/d3b-center/bixu-tracker/issues/1166), we said we could use WTCS. However, to keep the scale the same to allow batch-to-batch comparison, I ranked by WTCS, and then give them weights between 2 to 1 based on the rank. Please let me know whether this is acceptable. I can modify if not.